chainer/functions/normalization/batch_normalization.py

```diff
@@ -65,6 +309,8 @@ def __init__(self, eps=2e-5, mean=None, var=None, decay=0.9, axis=None):
             raise RuntimeError('axis must be int, tuple of int or None')
         self.axis = axis
 
+        self.bn_backend_selector = backend_selector
```

Comment:
How about `self._backend_selector`?
I think `bn` is redundant because this class is `BatchNormalization`.

Reply:
I used `bn` because there is already a `backend` there. 
https://github.com/chainer/chainer/blob/d2182099fcc59ac11daf358b496651730aed933c/chainer/functions/normalization/batch_normalization.py#L7
Might cause confusion

Reply:
Similarly, we have `communication_backend` in `chainermn`

Reply:
Hmm, I see. Then, how about using a different word than `backend`?

How about `impl`? This word is used in pooling implementation (`_CudnnPollingNDImpl` in `chainer/functions/pooling/pooilng_nd.py`). This class is semantically similar to `_BatchNormalizationBackend` class in this PR.

More specifically, my suggestion is:
* `_BatchNormalizationBackend` -> `_BatchNormalizationImpl`
* `BatchNormalization.bn_backend_selector` -> `BatchNormalization._impl_selector`
* `BatchNormalization(Grad).bn_backend` -> `BatchNormalization(Grad)._impl`

Reply:
Addressed in 4fae569